src/ngraph/op/fused/lstm_cell.hpp

```diff
@@ -0,0 +1,170 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "ngraph/autodiff/adjoints.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/op/util/activation_functions.hpp"
+#include "ngraph/op/util/fused_op.hpp"
+#include "ngraph/op/util/rnn_cell_base.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        ///
+        /// \brief      Class for lstm cell node.
+        ///
+        /// \note       It follows notation and equations defined as in ONNX standard:
+        ///             https://github.com/onnx/onnx/blob/master/docs/Operators.md#LSTM
+        ///
+        ///             Note this class represents only single *cell* and not whole LSTM *layer*.
+        ///
+        class LSTMCell : public util::FusedOp, public util::RNNCellBase
```

Comment:
@arogowie-intel , how do we define the whole lstm layer?, isnt most of the framwork have timesetp `{t, batch_size, input_size}` while defining for the inputs for lstm layer?

Reply:
From the nGraph perspective the whole LSTM layer isn't yet prepared (AFAIK). But we have defined ONNX version [here](https://github.com/NervanaSystems/ngraph/blob/arogowiec/fused_lstm_cell/src/ngraph/frontend/onnx_import/op/lstm.cpp). Please take a look at it's [inputs definitions](https://github.com/onnx/onnx/blob/master/docs/Operators.md#inputs-3---8). The first data input is indeed of shape: `[seq_length, batch_size, input_size]`.  And essentialy what we are doing [there](https://github.com/NervanaSystems/ngraph/blob/arogowiec/fused_lstm_cell/src/ngraph/frontend/onnx_import/op/lstm.cpp#L255-L358) is looping over `seq_length` axis of input and call `LSTMCell` on each batch.

Reply:
posted more discussion on #ngraph-lstm channel in slack.